python/cudf/cudf/core/reshape.py

```diff
@@ -169,14 +201,15 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
         raise ValueError("No objects to concatenate")
 
     objs = [obj for obj in objs if obj is not None]
-
     # Return for single object
     if len(objs) == 1:
         if ignore_index:
             result = cudf.DataFrame(
                 data=objs[0]._data.copy(deep=True),
                 index=cudf.RangeIndex(len(objs[0])),
             )
+            if axis == 1:
```

Comment:
When `axis=1` we are incorrectly letting the `axis=0` index behavior creep into the result. I've made changes to fix this and introduced test coverage for this case.